harness/src/test/java/org/jboss/pvt/harness/TestConfiguration.java

```diff
@@ -63,10 +112,81 @@ public Properties getAllConfiguration()
      * @return a new statement, which may be the same as {@code base},
      *         a wrapper around {@code base}, or a completely new Statement.
      */
-    @Override
-    public Statement apply( Statement base, Description description )
-    {
-//        throw new PVTSystemException( "NYI" );
-        return base;
+    public Statement apply(Statement base, Description description) {
+        return statement(base);
+    }
+
+    private Statement statement(final Statement base) {
+        return new Statement() {
+            @Override
+            public void evaluate() throws Throwable {
+                before();
+                try {
+                    base.evaluate();
+                } finally {
+                    after();
+                }
+            }
+        };
+    }
+
+    /**
+     * Override to set up your specific external resource.
+     *
+     * @throws Throwable if setup fails (which will disable {@code after}
+     */
+    protected void before() throws Throwable {
+        // load properties
+        if(properties.isEmpty()) {
+            String testFilename = System.getProperty(TEST_CONFIG_FILE_KEY);
+            try {
+                properties.load(new FileReader(testFilename));
+            } catch (Exception e) {
+                throw new PVTSystemException("File to load config file: " + testFilename, e);
+            }
+        }
+        prepareResources();
+    }
+
+    /**
+     * Override to tear down your specific external resource.
+     */
+    protected void after() {
+        // do nothing
+    }
+
+    protected synchronized void prepareResources() throws Exception{
+        String stagingServer = System.getProperty(TEST_STAGING_SERVER_KEY);
+
+        if(distribution == null) {
+            String distributionUrl = getConfiguration(PRODUCT_ZIP_URL_KEY).replace("{}", stagingServer);
+            String filename = distributionUrl.substring(distributionUrl.lastIndexOf("/") + 1);
```

Comment:
I am not convinced by this. This requires access to the VPN. The pull request builder won't work. It won't allow custom zip layouts to test in the validator.